glide-core/tests/test_socket_listener.rs

```diff
@@ -619,6 +619,83 @@ mod socket_listener {
         }
     }
 
+    #[rstest]
+    #[timeout(SHORT_CLUSTER_TEST_TIMEOUT)]
+    fn test_socket_pass_manual_route_by_address() {
+        let mut test_basics = setup_cluster_test_basics(false, true);
+
+        const CALLBACK_INDEX: u32 = 100;
+        let approx_message_length = 4 + APPROX_RESP_HEADER_LEN;
+        let mut buffer = Vec::with_capacity(approx_message_length);
+        let mut request = get_command_request(
+            CALLBACK_INDEX,
+            vec!["INFO".to_string(), "SERVER".to_string()],
+            RequestType::CustomCommand.into(),
+            false,
+        );
+        let mut routes = redis_request::Routes::default();
+        routes.set_simple_routes(redis_request::SimpleRoutes::Random);
```

Comment:
I think that it's more readable to get a node address from cluster slots / nodes, but if you do it this way please add comments explaining what you're doing - I was confused why do you test random route in the by_address test..

Reply:
with cluster slots/nodes you can take the host address too, not only the port

Reply:
> with cluster slots/nodes you can take the host address too, not only the port

but this is a local test, why does this matter? we know the host name.